lldb/source/Commands/CommandObjectMemory.cpp

```diff
@@ -885,6 +885,55 @@ class CommandObjectMemoryRead : public CommandObjectParsed {
 #define LLDB_OPTIONS_memory_find
 #include "CommandOptions.inc"
 
+static llvm::Error CopyExpressionResult(ValueObject &result,
+                                        DataBufferHeap &buffer) {
+  uint64_t value = result.GetValueAsUnsigned(0);
+  auto size_or_err = result.GetCompilerType().GetByteSize(nullptr);
+  if (!size_or_err)
+    return size_or_err.takeError();
+
+  switch (*size_or_err) {
+  case 1: {
+    uint8_t byte = (uint8_t)value;
+    buffer.CopyData(&byte, 1);
+  } break;
+  case 2: {
+    uint16_t word = (uint16_t)value;
+    buffer.CopyData(&word, 2);
+  } break;
+  case 4: {
+    uint32_t lword = (uint32_t)value;
+    buffer.CopyData(&lword, 4);
+  } break;
+  case 8: {
+    buffer.CopyData(&value, 8);
+  } break;
+  case 3:
+  case 5:
+  case 6:
+  case 7:
+    return llvm::createStringError("unknown type. pass a string instead");
```

Comment:
```suggestion
    return llvm::createStringError("unknown type. Pass a string instead");
```

Mildly confused about the "pass a string" advice. Is this user-visible and if yes, what's an example where this would be shown?

Reply:
(I also understand this isn't _your_ error message)

Reply:
Yea this is user visible. It would show exactly this error message:
```
unknown type. Pass a string instead
```

What it's trying to say is to use the `memory find -s` option instead. Could definitely be clearer

Reply:
Ah I see. How about?
```
Only expressions resulting in 1, 2, 4, or 8-byte-sized values are supported. For other pattern sizes the --string (-s) option may be used.
```